root/edit/details/EditWork.js

```diff
@@ -0,0 +1,186 @@
+/*
+ * @flow
+ * Copyright (C) 2020 Anirudh Jain
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import React from 'react';
+
+import DescriptiveLink from
+  '../../static/scripts/common/components/DescriptiveLink';
+import WordDiff from '../../static/scripts/edit/components/edit/WordDiff';
+import {INSERT, DELETE} from '../../static/scripts/edit/utility/editDiff';
+import Diff from '../../static/scripts/edit/components/edit/Diff';
+import DiffSide from '../../static/scripts/edit/components/edit/DiffSide';
+import FullChangeDiff from
+  '../../static/scripts/edit/components/edit/FullChangeDiff';
+import {commaOnlyListText} from
+  '../../static/scripts/common/i18n/commaOnlyList';
+
+type Props = {
+  ...EditT,
+  +data: {
+    +new: {
+      +attributes?: {
+        +[attributeName: string]: $ReadOnlyArray<WorkAttributeT>,
+      },
+      +languages: $ReadOnlyArray<string>,
+    },
+  },
+  +display_data: {
+    +attributes?: {
+      +[attributeName: string]: CompT<$ReadOnlyArray<string>>,
+    },
+    +comment?: CompT<string | null>,
+    +iswc?: CompT<string | null>,
+    +language?: CompT<LanguageT | null>,
+    +languages?: CompT<$ReadOnlyArray<string>>,
+    +name?: CompT<string>,
+    +type?: CompT<WorkTypeT | null>,
+    +work: WorkT,
+  },
+};
+
+const EditWork = ({edit}: {edit: Props}) => {
+  const display = edit.display_data;
+  const comment = display.comment;
+  const iswc = display.iswc;
+  const name = display.name;
+  const type = display.type;
+  const language = display.language;
+  const languages = display.languages;
+  const attributes = display.attributes ?? {};
```

Comment:
Checking `attributes ?` instead of `edit.data.new.attributes ?` below is probably sufficient so that it doesn't require a fallback.
```suggestion
  const attributes = display.attributes;
```